### PR TITLE
add clamp reference kernel with f16 and bf16 support

### PR DESCRIPTION
**Pull request by adstraw**

### CODE REVIEW

File: src/ngraph/op/fused/clamp.cpp [hunk 1]
```diff
@@ -18,12 +18,110 @@
...
+             count);
+            break;
+            TYPE_CASE(u64)
+            (arg,
+             out,
+             double_to_int<uint64_t>(min, ceil_func),
+             double_to_int<uint64_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(f16)(arg, out, static_cast<float16>(min), static_cast<float16>(max), count);
```
Comment (diyessi) [inline src/ngraph/op/fused/clamp.cpp:104]: Here the casts of large values should go to the fp16/bfloat16 infinities, but you need to test that it works.

File: test/util.cpp [hunk 2]
```diff
@@ -712,3 +712,56 @@ TEST(util, topological_sort_replace)
 
     EXPECT_TRUE(custom_sorter_used);
 }
```
Comment (diyessi) [inline test/util.cpp:714]: Need some bf16/fp16 cases, including with range out of bf16/fp16 non-infinity range.

File: src/ngraph/op/fused/clamp.cpp (hunk 1 above)
Comment (adstraw) [reply src/ngraph/op/fused/clamp.cpp:104]: Done.

File: test/util.cpp (hunk 2 above)
Comment (adstraw) [reply test/util.cpp:714]: Done.

**Pull request merged**